WeakLensingValidation/config_parser.py

```diff
@@ -0,0 +1,540 @@
+                raise ValueError(
+                    f"var_cat_history do not link to a list. Got: {var_name}"
+                )
+            if len(var_history) != len(path_output['path']):
+                raise ValueError(
+                    "Lenght of var_cat_history does not match number of "
+                    "catalogs."
+                    f"\nGot {len(var_history)} != {len(path_output['path'])}"
+                )
+            path_output['var_cat_history'] = var_history
+            return path_output
+        # We set a default list if not provided
+        else:
+            n_cat = len(path_output['path'])
+            path_output['var_cat_history'] = list(range(1, n_cat+1))
+            return path_output
+
+    def _parse_eval(self, column, name):
+        """parse eval
+
+        Parse column or path that will use the eval function.
+        Note that if the function contains variables, we first look for
+        general variables defined in the config file and then among the
+        catalog columns.
+
+        Parameters
+        ----------
+            column : dict
+                Dictionnary containing information about the eval
+            name: str
+                Name of variable to evaluate
+        Returns
+        -------
+            res : float
+                Result of eval fucntion
+            func : str
+                Function to evaluate with the column name to use from the
+                catalog
+            var_names: list
+                List of the column names to replace in the function
+        """
+        if 'type' not in column.keys():
+            raise ValueError(
+                f"Unreconized type for {name}, missing type"
+            )
+        if column['type'].lower() != 'eval':
+            raise ValueError(
+                f"Unreconized type for {name}, "
+                f"got {column['type']}"
+            )
+        if 'func' not in column.keys():
+            raise ValueError(
+                f"No function to evaluate for {name}"
+            )
+
+        # Copy the original string
+        func = copy.copy(column["func"])
+
+        # First check for variables
+        if '$' in func:
+            var_names = list(set(re.findall(r'\$(.*?)\$', func)))
+            for var_name in var_names:
+                if var_name in self._var.keys():
+                    func = func.replace(
+                        f"${var_name}$", f"{self._var[var_name]}"
+                    )
+                    var_names.remove(var_name)
+            if len(var_names) == 0:
+                try:
+                    res = eval(func)
+                except Exception as e:
+                    raise ValueError(
+                        f"Error while evaluating function: \n{func}\n"
+                        f"Got exception: \n{e}"
+                    )
+                return res
+            else:
+                return func, var_names
+        # Evaluate directly if no variables are found
```

Comment:
Probably better to have the following part in a separate function like `_eval` that only does the evaluation once everything is set.